source/server/admin/stats_request.h

```diff
@@ -26,22 +30,36 @@ class StatsRequest : public Admin::Request {
   // over all scopes for each type. This enables the complex chunking
   // implementation to pass the tests that capture the buffered behavior. There
   // is not a significant cost to this, but in a future PR we may choose to
-  // co-mingle the types. Note that histograms are groups together in the data
+  // co-mingle the types. Note that histograms are grouped together in the data
   // JSON data model, so we won't be able to fully co-mingle.
-  enum class Phase {
+  enum class PhaseName {
     TextReadouts,
     CountersAndGauges,
+    Counters,
+    Gauges,
     Histograms,
   };
 
+  struct Phase {
+    PhaseName phase;
+    std::string phase_label;
```

Comment:
Should the `phase_label` be looked up from a constant table based on the `PhaseName`? Or can the label vary for the `PhaseName`?